File: valid_parantheses.py
class Solution(object):
    def isValid(self, s):
        """
        :type s: str
        :rtype: bool
        """
        stack = []

        for x in s:
            if x in '([{':
                stack.append(x)
            else:
                if not stack or \
                    (x == ')' and stack[-1] != '(') or \
                    (x == ']' and stack[-1] != '[') or \
                    (x == '}' and stack[-1] != '{'):
                    return False
                stack.pop()
        return not stack

File: valid_parantheses.js
/**
 * @param {string} s
 * @return {boolean}
 */
var isValid = function (s) {
    let stack = [];

    for (let i = 0; i < s.length; i++) {
        let currentChar = s[i];

        if (currentChar == "(" || currentChar == "{" || currentChar == "[") {
            stack.push(currentChar);
        } else if (currentChar == ")" && stack[stack.length - 1] == "(") {
            stack.pop();
        } else if (currentChar == "}" && stack[stack.length - 1] == "{") {
            stack.pop();
        } else if (currentChar == "]" && stack[stack.length - 1] == "[") {
            stack.pop();
        } else {
            return false;
        }
    }

    return !stack.length;
};